### PR TITLE
Allow rehashing passwords when possible and needed

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Fixes https://github.com/symfony/maker-bundle/issues/382
Needs https://github.com/symfony/symfony/pull/31153

### CODE REVIEW

File: src/Maker/MakeUser.php [hunk 1]
```diff
@@ -32,6 +32,7 @@
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Input\InputOption;
 use Symfony\Component\Security\Core\Encoder\Argon2iPasswordEncoder;
+use Symfony\Component\Security\Core\Userr\PasswordUpgraderInterface;
```
Comment (przemyslaw-bogusz) [inline src/Maker/MakeUser.php]: There is double 'r' in User.

File: src/Resources/skeleton/security/UserProvider.tpl.php [hunk 2]
```diff
@@ -4,10 +4,11 @@
 
 use Symfony\Component\Security\Core\Exception\UnsupportedUserException;
 use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+<?= ($password_upgrader = interface_exists('Symfony\Component\Security\Core\User\PasswordUpgraderInterface')) ? "Symfony\Component\Security\Core\User\PasswordUpgraderInterface\n" : '' ?>
```
Comment (stof) [inline src/Resources/skeleton/security/UserProvider.tpl.php]: missing `use `

File: src/Maker/MakeUser.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Maker/MakeUser.php]: fixed thanks

File: src/Resources/skeleton/security/UserProvider.tpl.php (hunk 2 above)
Comment (nicolas-grekas) [reply src/Resources/skeleton/security/UserProvider.tpl.php]: fixed thanks

**Comment by weaverryan** [comment]
This is waiting for an upstream code merge

**Comment by weaverryan** [comment]
Thanks Nicolas!

**Pull request merged**